source/external/cJSON.c

```diff
@@ -1028,954 +1038,959 @@ static cJSON_bool print_object(const cJSON * const item, printbuffer * const out
+    {
+        buffer->offset--;
+    }
 
-  return buffer;
+    return buffer;
 }
 
 /* skip the UTF-8 BOM (byte order mark) if it is at the beginning of a buffer */
 static parse_buffer *skip_utf8_bom(parse_buffer * const buffer) // NOLINT
 {
-  if ((buffer == NULL) || (buffer->content == NULL) || (buffer->offset != 0))
-  {
-      return NULL;
-  }
+    if ((buffer == NULL) || (buffer->content == NULL) || (buffer->offset != 0))
+    {
+        return NULL;
+    }
 
-  if (can_access_at_index(buffer, 4) && (strncmp((const char*)buffer_at_offset(buffer), "\xEF\xBB\xBF", 3) == 0))
-  {
-      buffer->offset += 3;
-  }
+    if (can_access_at_index(buffer, 4) && (strncmp((const char*)buffer_at_offset(buffer), "\xEF\xBB\xBF", 3) == 0))
+    {
+        buffer->offset += 3;
+    }
 
-  return buffer;
+    return buffer;
 }
 
 CJSON_PUBLIC(cJSON *) cJSON_ParseWithOpts(const char *value, const char **return_parse_end, cJSON_bool require_null_terminated)
 {
-  size_t buffer_length;
+    size_t buffer_length;
 
-  if (NULL == value)
-  {
-      return NULL;
-  }
+    if (NULL == value)
+    {
+        return NULL;
+    }
 
-  /* Adding null character size due to require_null_terminated. */
-  buffer_length = strlen(value) + sizeof("");
+    /* Adding null character size due to require_null_terminated. */
+    buffer_length = strlen(value) + sizeof("");
 
-  return cJSON_ParseWithLengthOpts(value, buffer_length, return_parse_end, require_null_terminated);
+    return cJSON_ParseWithLengthOpts(value, buffer_length, return_parse_end, require_null_terminated);
 }
 
 /* Parse an object - create a new root, and populate. */
 CJSON_PUBLIC(cJSON *) cJSON_ParseWithLengthOpts(const char *value, size_t buffer_length, const char **return_parse_end, cJSON_bool require_null_terminated)
 {
-  parse_buffer buffer = { 0, 0, 0, 0, { 0, 0, 0 } };
-  cJSON *item = NULL;
-
-  if (value == NULL || 0 == buffer_length)
-  {
-      goto fail;
-  }
-
-  buffer.content = (const unsigned char*)value;
-  buffer.length = buffer_length;
-  buffer.offset = 0;
-  buffer.hooks = global_hooks;
-
-  item = cJSON_New_Item(&global_hooks);
-  if (item == NULL) /* memory fail */
-  {
-      goto fail;
-  }
-
-  if (!parse_value(item, buffer_skip_whitespace(skip_utf8_bom(&buffer))))
-  {
-      /* parse failure. ep is set. */
-      goto fail;
-  }
-
-  /* if we require null-terminated JSON without appended garbage, skip and then check for a null terminator */
-  if (require_null_terminated)
-  {
-      buffer_skip_whitespace(&buffer);
-      if ((buffer.offset >= buffer.length) || buffer_at_offset(&buffer)[0] != '\0')
-      {
-          goto fail;
-      }
-  }
-  if (return_parse_end)
-  {
-      *return_parse_end = (const char*)buffer_at_offset(&buffer);
-  }
-
-  return item;
+    parse_buffer buffer = { 0, 0, 0, 0, { 0, 0, 0 } };
+    cJSON *item = NULL;
+
+    /* reset error position */
+    global_error.json = NULL;
+    global_error.position = 0;
+
```

Comment:
global_error is removed in our copy of cJson due to race conditions https://github.com/awslabs/aws-c-common/pull/972/files